code/datums/helper_datums/teleport.dm

```diff
@@ -202,3 +193,19 @@
 	var/list/turfs = get_teleport_turfs(center, precision)
 	if (length(turfs))
 		return pick(turfs)
+
+/// Validates that the teleport being attempted is valid or not
+/proc/check_teleport_valid(atom/tele_atom, atom/destination, area/from_area, area/to_area, turf/from_turf, turf/to_turf, channel)
```

Comment:
Expand `tele_atom` to `teleporting_atom` or similar, I think this is ambiguous with `teleporter_atom`

Reply:
It is also not obvious at all why we pass a destination, AND a to_area, AND a to_turf. please document the parameters, and/or eliminate the need for them

Reply:
Cleaned up